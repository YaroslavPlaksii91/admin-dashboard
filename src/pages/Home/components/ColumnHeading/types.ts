export type ColumnHeadingProps = {
  onClick: () => void;
  size: number;
  name: string;
  title: string;
  sortKey: string;
  sortDirection: string;
};
