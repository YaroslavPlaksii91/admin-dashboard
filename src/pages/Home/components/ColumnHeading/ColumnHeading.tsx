import { FC } from 'react';
import { Grid } from '@mui/material';
import NorthIcon from '@mui/icons-material/North';
import SouthIcon from '@mui/icons-material/South';

import { ColumnHeadingProps } from './types';

export const ColumnHeading: FC<ColumnHeadingProps> = ({
  onClick,
  size,
  name,
  title,
  sortKey,
  sortDirection,
}) => {
  return (
    <Grid item xs={size} onClick={onClick}>
      {title}{' '}
      {sortKey === name &&
        (sortDirection === 'asc' ? (
          <NorthIcon fontSize="small" sx={{ verticalAlign: 'top' }} />
        ) : (
          <SouthIcon fontSize="small" sx={{ verticalAlign: 'top' }} />
        ))}
    </Grid>
  );
};
