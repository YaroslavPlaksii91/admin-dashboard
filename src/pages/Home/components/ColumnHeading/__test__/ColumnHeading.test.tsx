import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import { ColumnHeading } from '../ColumnHeading';

describe('ColumnHeading component', () => {
  afterEach(cleanup);

  const onClick = jest.fn();

  it('should be rendered', () => {
    const { asFragment } = render(
      <ColumnHeading
        onClick={onClick}
        size={4}
        name="columnName"
        title="Column Title"
        sortKey="columnName"
        sortDirection="asc"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with ascending sort', () => {
    render(
      <ColumnHeading
        onClick={onClick}
        size={4}
        name="columnName"
        title="Column Title"
        sortKey="columnName"
        sortDirection="asc"
      />,
    );

    expect(screen.getByText('Column Title')).toBeInTheDocument();
    expect(screen.getByTestId('NorthIcon')).toBeInTheDocument();
  });

  it('renders correctly with descending sort', () => {
    render(
      <ColumnHeading
        onClick={onClick}
        size={4}
        name="columnName"
        title="Column Title"
        sortKey="columnName"
        sortDirection="desc"
      />,
    );

    expect(screen.getByText('Column Title')).toBeInTheDocument();
    expect(screen.getByTestId('SouthIcon')).toBeInTheDocument();
  });

  it('calls onClick when clicked', () => {
    render(
      <ColumnHeading
        onClick={onClick}
        size={4}
        name="columnName"
        title="Column Title"
        sortKey="columnName"
        sortDirection="asc"
      />,
    );

    fireEvent.click(screen.getByText('Column Title'));

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
