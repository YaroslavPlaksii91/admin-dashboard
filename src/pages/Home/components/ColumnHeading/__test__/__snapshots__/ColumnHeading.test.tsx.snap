// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnHeading component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-13i4rnv-MuiGrid-root"
  >
    Column Title 
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1j6x8sl-MuiSvgIcon-root"
      data-testid="NorthIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m5 9 1.41 1.41L11 5.83V22h2V5.83l4.59 4.59L19 9l-7-7-7 7z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
